src/mono/netcore/System.Private.CoreLib/src/System/Threading/TimerQueue.Browser.cs

```diff
@@ -0,0 +1,127 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Threading
+{
+    // Runtime interface
+    internal static class WasmRuntime
+    {
+        private static Dictionary<int, Action> callbacks = new Dictionary<int, Action>();
```

Comment:
Also just to re-confirm, this isn't thread-safe, and this is all with the expectation that it's only ever used from a single thread.  I wonder how much clean-up we're going to need to do and whether we'll be able to successfully audit everything well enough when that expectation changes.

Reply:
Removed the locking code for now.